### PR TITLE
`Quiz exercises`: Adjust user interface of quiz header and footer

### PR DESCRIPTION
**Pull request by rabeatwork**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

The previous footer had issues, especially on smaller screens. 

### Description

Rearranged some information and buttons. Removed duplicated information (e.g. practice or preview). Stick footer to the bottom. Align the design in general and refactored some code. 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- Multiple Quiz Exercises (Synchrinised, Barched, Individual)

1. Log in to Artemis
2. Navigate to Course Overview --> Exercises
3. Start different Quiz Exercises in different Modes (Preview, Pracitce, Live, View Results)
4. Verify that everything works as expected 

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Before:
Live:
Screenshot 2024-11-12 at 15 03 39
Screenshot 2024-11-12 at 15 03 55
Practice:
Screenshot 2024-11-12 at 15 03 01

After: 
Live:
Screenshot 2024-11-12 at 15 00 37
assets/717221e5-35f9-4e5a-baef-47c9bd3faaf6">

Practice:
Screenshot 2024-11-12 at 15 01 07

## Release Notes

- **New Features**
	- Enhanced layout and structure for the quiz participation interface.
	- Improved modularity for mode indicators and consolidated display logic for quiz instructions and results.
	- Added routing capabilities to differentiate between management and standard views.

- **Bug Fixes**
	- Streamlined connection status display logic for clarity.

- **Style**
	- Updated styling for quiz elements, including time warnings and footer sections, focusing on a more streamlined approach.
	- Introduced new CSS classes for improved layout flexibility and sticky positioning.

- **Localization**
	- Simplified phrasing for auto-save messages in both German and English localization files.

### CODE REVIEW

**Review by sawys777 — APPROVED** [review]
Tested on TS1, everything worked as expected, the quiz interface looks exactly like the ones in the "after" screenshots.

**Review by HawKhiem — COMMENTED** [review]
Tested on TS5. Everything mentioned works. One thing I found strange was Artemis somehow allows a quiz in Batched and Individual mode

Screenshot 2024-11-16 200615
Screenshot 2024-11-16 200937

 to have a due date sooner than the working time. For example the quiz is visible from 8pm, the working time is 10 minutes, but it is still allowed that the due date set to 8:05 pm

**Review by janthoXO — DISMISSED** [review]
for me the header in the preview was not working correctly. the exercise header is above the Breadcrumb Navigation which looks not correct

https://github.com/user-attachments/assets/c76041d7-886c-4ff3-b050-e03f3cb90b0a

**Comment by janthoXO** [comment]
also is there a reason why the footer in the edit view has that much padding?
image

**Review by flbrgit — APPROVED** [review]
Tested on TS3. The features work as described, only on very small screens the modal window after submitting an exercise is not entirely visible/scrollable
Screenshot 2024-11-19 181217
It is also possible to set the end date of a quiz in the past also for new quizzes
Screenshot 2024-11-19 181805

**Review by JerroyTan — CHANGES_REQUESTED** [review]
[Tested on TS3]
Footer works just fine, only goes above content when zoomed to very large.

I however encountered problems with the header.
When showing the quiz and opening the notification tab, the header of the quiz is not grayed out to the back and covers the notification tab.
image
image
I do not think this behavior is intended

**Review by JerroyTan — APPROVED** [review]
[Tested on TS6]
Works just as described it should. Look good to me

**Review by sawys777 — APPROVED** [review]
Tested on TS 6, Reapprove.

**Review by flbrgit — APPROVED** [review]
Tested on TS3, reapprove

**Review by HawKhiem — APPROVED** [review]
Reapprove

File: src/main/webapp/app/exercises/quiz/participate/quiz-participation.component.ts [hunk 1]
```diff
@@ -132,6 +133,7 @@ export class QuizParticipationComponent implements OnInit, OnDestroy {
         private quizExerciseService: QuizExerciseService,
         private participationService: ParticipationService,
         private route: ActivatedRoute,
+        private router: Router,
```
Comment (florian-glombik) [inline src/main/webapp/app/exercises/quiz/participate/quiz-participation.component.ts:136]: Was it an active decision not use the more explicit _inject_ here? 

`private readonly router = inject(Router);`

File: src/main/webapp/app/shared/connection-status/connection-status.component.html [hunk 2]
```diff
@@ -1,19 +1,12 @@
 <div class="connection-status" [ngClass]="{ disconnected: disconnected }">
-    @if (!isExamMode) {
-        <fa-icon [icon]="faCircle" size="sm" />
-    }
```
Comment (florian-glombik) [inline src/main/webapp/app/shared/connection-status/connection-status.component.html:4]: Why don't we need this anymore?

File: src/main/webapp/app/shared/connection-status/connection-status.component.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
 .connection-status {
-    font-size: 0.85rem;
+    font-size: 0.75rem;
```
Comment (florian-glombik) [inline src/main/webapp/app/shared/connection-status/connection-status.component.scss:2]: As it appears that the font size shall be consistent, would it make sense to define a shared variable for the `font-size`?

**Review by florian-glombik — APPROVED** [review]
Thanks for the introduction of the templates and re-structuring which makes the template easier to read

I left some minor comments where I do not have a strong opinion, I see more work for this component but not within the scope of this PR:
* I think that it might be good to divide the `quiz-participation.component.html` into further smaller components (e.g. another on vor the _Overlay section_ and one for the _Step Wizard_)
* The complexity of the template could also additionally be reduced by adding computed signals, which would also require the introduction of signals 😅

File: src/main/webapp/app/shared/connection-status/connection-status.component.html (hunk 2 above)
Comment (rabeatwork) [reply src/main/webapp/app/shared/connection-status/connection-status.component.html:4]: Unified design :)

File: src/main/webapp/app/shared/connection-status/connection-status.component.scss (hunk 3 above)
Comment (rabeatwork) [reply src/main/webapp/app/shared/connection-status/connection-status.component.scss:2]: Yes, we could do this in a separate PR for multiple occurrences

**Comment by rabeatwork** [comment]
> Thanks for the introduction of the templates and re-structuring which makes the template easier to read
> 
> I left some minor comments where I do not have a strong opinion, I see more work for this component but not within the scope of this PR:
> 
> * I think that it might be good to divide the `quiz-participation.component.html` into further smaller components (e.g. another on vor the _Overlay section_ and one for the _Step Wizard_)
> * The complexity of the template could also additionally be reduced by adding computed signals, which would also require the introduction of signals 😅

I totally agree, I just wanted to approach it in smaller steps.

**Review by krusche — APPROVED** [review]
The new header looks much better 👍

**Pull request merged**